src/se/vidstige/jadb/JadbDevice.java

```diff
@@ -40,6 +40,12 @@ public String getSerial() {
         return serial;
     }
 
+    public String getState() throws IOException, JadbException {
+        Transport transport = this.getTransport();
+        this.send(transport, "host-serial:" + this.serial + ":get-state");
+        return transport.readString();
+    }
```

Comment:
* this should have gone it its own commit.
* Add test case for this in `RealDeviceTests`